src/samplers/array.js

```diff
@@ -1,16 +1,24 @@
 import { traverse } from '../traverse';
 export function sampleArray(schema, options = {}, spec, context) {
   const depth = (context && context.depth || 1);
-
-  let arrayLength = Math.min(schema.maxItems != undefined ? schema.maxItems : Infinity, schema.minItems || 1);
   // for the sake of simplicity, we're treating `contains` in a similar way to `items`
-  const items = schema.items || schema.contains;
+  const items = schema.prefixItems || schema.items || schema.contains;
+
+  let arrayLength = Math.min(
+    schema.maxItems != undefined ? schema.maxItems : Infinity,
+    schema.minItems != undefined
+      ? schema.minItems
+      : Array.isArray(items)
+      ? items.length
+      : 1,
+  );
+
```

Comment:
Maybe like this instead?

```suggestion
  let arrayLength = Array.isArray(items) ? items.length:  Math.min(
    schema.maxItems != undefined ? schema.maxItems : Infinity,
    schema.minItems != undefined
      ? schema.minItems
      : 1,
  );

```